Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -2056,6 +2072,14 @@ theorem le_mk_iff_exists_subset {c : Cardinal} {α : Type u} {s : Set α} :
   rw [le_mk_iff_exists_set, ← Subtype.exists_set_subtype]
   apply exists_congr; intro t; rw [mk_image_eq]; apply Subtype.val_injective
 
+@[simp]
+theorem mk_range_inl {α : Type u} {β : Type v} : #(range (@Sum.inl α β)) = lift.{v} #α := by
```

Comment:
This one looks like it should be a composition of two simpler lemmas. It looks like those should be `range (@Sum.inl α β) = (Set.univ : Set α)`, and `#(Set.univ : Set α) = #α` (maybe with some `lift`s thrown in). Then those should both be simp lemmas, and this lemma wouldn't need to exist at all.

Reply:
The issue is that the first lemma you give doesn't typecheck, as `range (@Sum.inl α β) : Set (α ⊕ β)`. We instead have the much uglier [`Set.range_inl`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Set/Image.html#Set.range_inl) which doesn't make much sense as a `simp` normal form.